File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        if(s.empty()) return true;
        
        int l = 0;
        int r = s.length() - 1;

        while(l <= r){
            if(!isalnum(s[l]) || s[l] == ' '){
                l++;
                continue;
            }
            if(!isalnum(s[r]) || s[r] == ' '){
                r--;
                continue;
            }
            if(tolower(s[l++]) != tolower(s[r--]))
                return false;
        }

        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.go
import "strings"

func isAlnum(c byte) bool {
    return (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9')
}

func isPalindrome(s string) bool {
    lo := 0
    hi := len(s) - 1

    str := strings.ToLower(s)
    for hi > lo{
        if !isAlnum(str[lo]){
            lo++
            continue
        }
        if !isAlnum(str[hi]){
            hi--
            continue
        }
        if str[hi] != str[lo]{
            return false
        }

        lo++
        hi--
    }

    return true
}